cdap-app-fabric/src/test/java/co/cask/cdap/internal/AppFabricClient.java

```diff
@@ -239,6 +248,51 @@ public Instances getFlowletInstances(String namespaceId, String applicationId, S
     return schedules;
   }
 
+  public WorkflowTokenDetail getWorkflowToken(String namespaceId, String appId, String wflowId, String runId,
+                                              @Nullable WorkflowToken.Scope scope,
+                                              @Nullable String key) throws NotFoundException {
+    MockResponder responder = new MockResponder();
+    String uri = String.format("%s/apps/%s/workflows/%s/runs/%s/token",
+                               getNamespacePath(namespaceId), appId, wflowId, runId);
+    HttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, uri);
+    if (scope == null) {
+      scope = WorkflowToken.Scope.USER;
+    }
+    if (key == null) {
+      key = "";
+    }
+    workflowHttpHandler.getWorkflowToken(request, responder, namespaceId, appId, wflowId, runId, scope.name(), key);
+
+    Type workflowTokenDetailType = new TypeToken<WorkflowTokenDetail>() { }.getType();
+    WorkflowTokenDetail workflowTokenDetail = responder.decodeResponseContent(workflowTokenDetailType, GSON);
+    verifyResponse(HttpResponseStatus.OK, responder.getStatus(), "Getting workflow token failed");
+    return workflowTokenDetail;
+  }
+
+  public WorkflowTokenNodeDetail getWorkflowToken(String namespaceId, String appId, String wflowId, String runId,
```

Comment:
hmm, just went by the pattern in this class. Thought it would it be better to refactor this class at once to use ``Id`` classes, as opposed to using them in just a couple of methods.